### PR TITLE
Improve documentation for NDEFMessage.

### PR DESCRIPTION
**Pull request by jpmedley**
> Issue number that this PR fixes (if any). For example: 'Fixes #987654321'

NA

> What was wrong/why is this fix needed? (quick summary only)

This  PR adds additional information related to the interface.

> Anything else that could help us review it

No

### CODE REVIEW

File: files/en-us/web/api/ndefmessage/ndefmessage/index.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+<h3 id="Parameters">Parameters</h3>
+
+
+<dl>
+  <dt><code>records</code></dt>
+  <dd>An array of objects with the following members:
+    <dl>
+      <dt><code>data</code> {{optional_inline}}</dt>
+      <dd>Contains the data to be transmitted, one of a string, a {{domxref("BufferSource")}}, or an array of nested records.</dd>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefmessage/ndefmessage/index.html:27]:
```suggestion
      <dd>Contains the data to be transmitted; one of a string, a {{domxref("BufferSource")}}, or an array of nested records.</dd>
```

File: files/en-us/web/api/ndefmessage/ndefmessage/index.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  </dd>
+</dl>
+
+<h2 id="Specifications">Specifications</h2>
+
+<p>{{Specifications}}</p>
+
+<h2 id="Browser_compatibility">Browser compatibility</h2>
+
+<div class="hidden">The compatibility table in this page is generated from structured data. If you'd like to contribute to the data, please check out <a href="https://github.com/mdn/browser-compat-data">https://github.com/mdn/browser-compat-data</a> and send us a pull request.</div>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefmessage/ndefmessage/index.html:66]:
```suggestion
```
I think we don’t need to add this boilerplate any longer.

File: files/en-us/web/api/ndefmessage/records/index.html [hunk 3]
```diff
@@ -23,6 +23,32 @@ <h3 id="Value">Value</h3>
 <p>A list of {{DOMxRef("NDEFRecord")}} object that represent data recorded in the message.
 </p>
 
+<h2 id="Examples">Examples</h2>
+
+<p>The following example shows how to read the contents of an NDEF message. It first sets up an event handler for {{domxrex("NDEFReader.onreading")}}, which is passed an instance of {{domxref("NDEFReadingEvent")}}. An <code>NDEFMessage</code> object is returned from {{domxref("NDEFReadingEvent.message")}}. It loops through <code>message.records</code> and processes each record based on its message type. The data member is a {{domxref('DataView")}}, which allows handling data encoded in UTF-16.</p>
+
+<p>This text should be replaced with a brief description of what the example demonstrates.</p>
+
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefmessage/records/index.html:31]:
```suggestion
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by jpmedley** [comment]
Please take another look.

**Pull request merged**